tests/samples_tests/CMakeLists.txt

```diff
@@ -6,4 +6,11 @@ cmake_minimum_required(VERSION 3.13)
 
 project(samples_tests)
 
-install(DIRECTORY smoke_tests/ DESTINATION tests/smoke_tests COMPONENT tests EXCLUDE_FROM_ALL)
+install(DIRECTORY smoke_tests/ DESTINATION tests/smoke_tests COMPONENT tests EXCLUDE_FROM_ALL PATTERN "requirements.txt" EXCLUDE)
+install(FILES ../constraints.txt DESTINATION tests COMPONENT tests EXCLUDE_FROM_ALL)
+
+file(READ smoke_tests/requirements.txt REQUIREMENTS_REPO)
+string(REPLACE "-c ../../constraints.txt" "-c ../constraints.txt" REQUIREMENTS_TMP ${REQUIREMENTS_REPO})
+file(WRITE ./requirements.txt ${REQUIREMENTS_TMP})
```

Comment:
it's bad practice to write inside source folder.
Use `CMAKE_CURRENT_BINARY_DIR` to replicate current sub-folder to build tree (binary tree)

Reply:
Done, please take a look if it's alright now.